File: 0976-largest-perimeter-triangle/0976-largest-perimeter-triangle.cpp
class Solution {
public:
    int largestPerimeter(vector<int>& nums) {
        sort(nums.rbegin(),nums.rend());
        int res=0;
        for(int i=2;i<nums.size();i++)
        {
            if(nums[i]+nums[i-1]>nums[i-2])
            {
                return nums[i-2]+nums[i-1]+nums[i];
            }
        }
        return res;
    }
};

File: 0976-largest-perimeter-triangle/0976-largest-perimeter-triangle.java
class Solution {
    public int largestPerimeter(int[] nums) {
        Arrays.sort(nums);
        int perimeter = 0;
        for(int i=2;i<nums.length;i++)
        {
            int a = nums[i], b = nums[i-1], c = nums[i-2];
            if((a+b)>c && (a+c)>b && (b+c)>a)
            {
                perimeter = Math.max(perimeter, a+b+c);
            }
        }
        return perimeter;
    }
}